### PR TITLE
network: fix MTU dialog error in Automatic mode

### LINKED ISSUE #19130: Saving MTU with "Automatic" selected results in an error "MTU must be a positive number"
### Explain what happens

1. Open the MTU dialog in network interface page (e.g. ethernet, bond etc)
2. While the option "Automatic" is selected click "Save"

image

(It doesn't happen when switching from previously set Custom value to Automatic)

### Version of Cockpit

main branch

### Where is the problem in Cockpit?

Networking

### Server operating system

Fedora

### Server operating system version

38

### What browsers are you using?

Brave

### System log

_No response_

### PR DESCRIPTION
**Pull request by subhoghoshX**
Fixes #19130

### CODE REVIEW

File: pkg/networkmanager/mtu.jsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const MtuDialog = ({ connection, dev, settings }) => {
     const [mtu, setMtu] = useState(settings.ethernet.mtu ? settings.ethernet.mtu : '');
 
     const onSubmit = (ev) => {
-        const mtuNew = parseInt(mtu, 10);
+        const mtuNew = mode == 'auto' ? 0 : parseInt(mtu, 10);
```
Comment (cockpituous) [inline pkg/networkmanager/mtu.jsx:41]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19156-20230801-065114-15e72e1d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
Thanks!

File: pkg/networkmanager/mtu.jsx (hunk 1 above)
Comment (martinpitt) [reply pkg/networkmanager/mtu.jsx:41]: That means that we don't have a test which exercises that very thing (setting MTU to "auto"). That would be nice to add before we land this.

Comment (jelly) [reply pkg/networkmanager/mtu.jsx:41]: Sorry, I merged this before I noticed your review comment. @subhoghoshX mind adding a test?

Comment (subhoghoshX) [reply pkg/networkmanager/mtu.jsx:41]: Oops. @jelly no issues, adding it to my todo, thanks for mentioning.

Question: now I wonder, is there a way to run the coverage bot locally on a commit or something?

**Pull request merged**